src/_gettsim/policy_environment_postprocessor.py

```diff
@@ -771,14 +562,20 @@ def aggregate_by_p_id_func(column, p_id_to_aggregate_by, p_id_to_store_by):
         else:
             raise ValueError(f"Aggr {agg_specs['aggr']} is not implemented.")
 
-    return aggregate_by_p_id_func
+    return DerivedFunction(
+        aggregate_by_p_id_func,
+        function_name=agg_col,
+    )
```

Comment:
```suggestion
        derived_from=derived_from,
    )
```

In if/else block above, incorporate:

```py
    if agg_specs["aggr"] == "count":
        derived_from = p_id_to_aggregate_by
    else:
        derived_from = (column, p_id_to_aggregate_by)
```